contrib/99designs/gqlgen/tracer_test.go

```diff
@@ -0,0 +1,149 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package gqlgen
+
+import (
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+
+	"github.com/99designs/gqlgen/client"
+	"github.com/99designs/gqlgen/example/todo"
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/99designs/gqlgen/handler"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestImplementsTracer(t *testing.T) {
+	var _ graphql.Tracer = (*gqlTracer)(nil)
+}
+
+func newTodoClient(t graphql.Tracer) *client.Client {
+	c := client.New(handler.GraphQL(
+		todo.NewExecutableSchema(todo.New()),
+		handler.Tracer(t),
+	))
+	return c
+}
+
+func TestOptions(t *testing.T) {
+	for name, tt := range map[string]struct {
+		tracerOpts []Option
+		test       func(assert *assert.Assertions, root mocktracer.Span)
+	}{
+		"default": {
+			test: func(assert *assert.Assertions, root mocktracer.Span) {
+				assert.NotNil(root)
+				assert.Equal(root.Tag(ext.ResourceName), "gqlgen.operation")
```

Comment:
Let's check the operation name here, not the resource. The resource isn't even set.